include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -2603,7 +2605,10 @@ __pattern_partial_sort_copy(__parallel_tag<_IsVector>, _ExecutionPolicy&& __exec
                     // 1. Copy elements from input to raw memory
                     for (_T1* __k = __i; __k != __j; ++__k, ++__it)
                     {
-                        ::new (__k) _T2(*__it);
+                        if constexpr (std::is_trivially_copy_constructible_v<_T2>)
+                            *__k = *__it;
+                        else
+                            ::new (__k) _T2(*__it);
```

Comment:
The value of this change is not clear to me. As far as I understand, if the type is trivially copyable, it can be copied bitwise (e.g. with `std::memcpy`) instead of calling the copy constructor directly. In this particular case, we just replace the copy constructor call with the copy assignment call. May be I simply don't understand why it is better. Could you please explain in details if you find it useful?

Reply:
As I described at this PR, I simple trying to avoid constructor/destructor calls when it's not required.
In this case we simple make copy assignment call without constructor call - I hope it's should work faster then constructor call.

Reply:
It seems like this change introduced more problems then benefits. According to the standard, the class with deleted copy assignment operator still considered trivially copyable. Consider this example: https://godbolt.org/z/ofPYaEMW9.

Reply:
Fixed

Reply:
We already have this approach in our code: https://github.com/oneapi-src/oneDPL/blob/8b9a2dd9a347c0255e4759fa93f6738e5d103340/include/oneapi/dpl/pstl/algorithm_impl.h#L1498